qemu/tests/pci_bridge_hotplug_unplug.py

```diff
@@ -0,0 +1,142 @@
+import logging
+import time
+
+from virttest import error_context
+from virttest import utils_test
+
+
+def disk_hotplug(test, vm, params, image, drive_format, parent_bus):
+    """
+    Hotplug new disk.
+
+    :param test: QEMU test object
+    :param vm: VM object
+    :param params: Dictionary with the test parameters
+    :param image: Name of the new disk
+    :param drive_format: Drive subsystem type (virtio, scsi, usb3)
+    :param parent_bus: Bus(es), in which this device is plugged in
+    """
+
+    if drive_format not in ('virtio', 'scsi-hd', 'usb3'):
+        raise NotImplementedError()
+
+    image_params = params.object_params(image)
+    image_params["drive_format"] = drive_format
+    devices = []
+
+    if drive_format == 'usb3':
+        usbc_params = {'usb_type': 'nec-usb-xhci'}
+        devices = vm.devices.usbc_by_params(drive_format, usbc_params, pci_bus={'aobject': parent_bus})
+
+    devices += vm.devices.images_define_by_params(image, image_params, 'disk',
+                                                  None, False, None, pci_bus={'aobject': parent_bus})
+
+    for device in devices:
+        vm.devices.insert(device)
+        device.hotplug(vm.monitor)
+
+    ver_out = devices[-1].verify_hotplug("", vm.monitor)
+    if not ver_out:
+        test.fail("%s disk is not in qtree after hotplug!" % drive_format)
+
+    if drive_format == 'virtio':
+        return [devices[-1]]
+    else:
+        return devices[::2]
+
+
+def check_all_hotplugged_disks(test, params, env, image_num):
+    """
+    Check all hotplugged disks.
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    error_context.context("Check 'info block'!", logging.info)
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+
+    monitor_info_block = vm.monitor.info_block(False)
+    if image_num != len(monitor_info_block.keys()):
+        test.fail("The amount of block devices in 'info block' is incorrect!")
```

Comment:
Better to put the two number and 'info block' detail in message content.

Reply:
Modified, Thanks!